packages/master-detail-layout/src/vaadin-master-detail-layout-transition-styles.js

```diff
@@ -12,47 +12,82 @@ export const transitionStyles = css`
     view-transition-name: vaadin-master-detail-layout-overlay-horizontal-detail-add;
   }
 
-  ::view-transition-group(vaadin-master-detail-layout-overlay-horizontal-detail-add) {
+  vaadin-master-detail-layout[dir='rtl'][overlay][orientation='horizontal'][transition='add']::part(detail) {
```

Comment:
Can we use a custom CSS property for this? Depending on the dir attribute it would have the values `[1, -1]`, and then gets multiplied with the translate value.

Reply:
Good idea, thanks. I'll try this approach.

Reply:
Updated to use custom CSS property. Note, we need it to be declared on the `html` apparently since the animation applies to global pseudo elements (also, view transitions won't work when MDL is wrapped in shadow root).

As a result, we probably don't need `SlotStylesMixin` after all but IMO it's ok to keep it for now.